client/src/http_default.rs

```diff
@@ -0,0 +1,197 @@
+use std::{borrow::Borrow, net::TcpStream};
+
+use attohttpc::{
+    body as atto_body, header::HeaderName, RequestBuilder as AttoHttpRequestBuilder,
+    Response as AttoHttpResponse,
+};
+use eyre::{eyre, Error, Result, WrapErr};
+use tungstenite::{stream::MaybeTlsStream, WebSocket};
+pub use tungstenite::{Error as WebSocketError, Message as WebSocketMessage};
+
+use crate::http::{Headers, Method, RequestBuilder, Response};
+
+type Bytes = Vec<u8>;
+
+trait AttoHttpReqExt: Sized {
+    fn set_headers(self, headers: Headers) -> Result<Self>;
+}
+
+impl AttoHttpReqExt for AttoHttpRequestBuilder<atto_body::Bytes<Vec<u8>>> {
+    fn set_headers(mut self, headers: Headers) -> Result<Self> {
+        for (h, v) in headers {
+            let h = HeaderName::from_bytes(h.as_ref())
+                .wrap_err_with(|| format!("Failed to parse header name {}", h))?;
+            self = self.header(h, v);
+        }
+        Ok(self)
+    }
+}
+
+trait HttpReqExt: Sized {
+    fn set_headers(self, headers: Headers) -> Result<Self>;
+}
```

Comment:
I think you can use only one ext trait named `SetHeadersExt` (so it is named in honor of new functional) and use it for both `AttoHttpRequestBuilder` and `http::request::Builder`

Reply:
Refactored :heavy_check_mark: